arch/risc-v/src/esp32c3/Kconfig

```diff
@@ -261,6 +262,29 @@ config ESP32C3_SPIFLASH
 	select MTD_BYTE_WRITE
 	select MTD_PARTITION
 
+config ESP32C3_SPI_LCD
+	bool "Generic SPI LCD Module"
+	default n
+	select ESP32C3_SPI2
+	select SPI_CMDDATA
+	select ESP32C3_SPI_SWCS
+	select SPI_SWAPBYTES
+
+if ESP32C3_SPI_LCD
+
+config SPI_SWAPBYTES
+	bool "Swap high and low bytes during 16bit SPI send"
```

Comment:
Shouldn't this swapping be performed at application-level?
See that LVGL has an option for this:

https://github.com/apache/incubator-nuttx-apps/blob/master/graphics/lvgl/Kconfig#L119-L122

Reply:
Yes you're right, that is being used for correct display data. This config here is for swapping the bytes of command data, which is not really necessary if the command data was sent as 8-bit in the first place. So I'm a little confused here as whether to add swapping to the SPI driver or just change the drivers to 8bit SPI. (I believed someone use 16bit for a reason?)

Reply:
> Yes you're right, that is being used for correct display data. This config here is for swapping the bytes of command data, which is not really necessary if the command data was sent as 8-bit in the first place.

Agree. The SPI display drivers are responsible for configuring the SPI width before sending command and data.

> So I'm a little confused here as whether to add swapping to the SPI driver or just change the drivers to 8bit SPI. (I believed someone use 16bit for a reason?)

Sorry, I couldn't follow. But if there is any command being erroneously sent as 16-bit, it should be fixed at the peripheral driver. And the if the data needs to be swapped, the application should be the one responsible for it.

The SPI controller driver shouldn't have this kind of knowledge.